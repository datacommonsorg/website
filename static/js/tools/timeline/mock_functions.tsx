/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

jest.mock("axios");

import axios from "axios";
import * as d3 from "d3";
import { when } from "jest-when";

import { drawGroupLineChart } from "../../chart/draw";
import { stringifyFn } from "../../utils/axios";

export function axios_mock(): void {
  // Mock all the async axios call.
  axios.get = jest.fn();
  axios.post = jest.fn();

  // get statsvar properties Median_Age_Person
  when(axios.get)
    .calledWith("/api/stats/stats-var-property?dcid=Median_Age_Person")
    .mockResolvedValue({
      data: {
        Median_Age_Person: {
          md: "",
          mprop: "age",
          pt: "Person",
          pvs: {},
          title: "Age",
        },
      },
    });

  // get statsvar properties Count_Person
  when(axios.get)
    .calledWith("/api/stats/stats-var-property?dcid=Count_Person")
    .mockResolvedValue({
      data: {
        Count_Person: {
          md: "",
          mprop: "count",
          pt: "Person",
          pvs: {},
          title: "Population",
        },
      },
    });

  // get statsVar info of Median_Age_Person and Count_Person
  when(axios.get)
    .calledWith(
      "/api/stats/stats-var-property?dcid=Median_Age_Person&dcid=Count_Person"
    )
    .mockResolvedValue({
      data: {
        Median_Age_Person: {
          md: "",
          mprop: "age",
          pt: "Person",
          pvs: {},
          title: "Age",
        },
        Count_Person: {
          md: "",
          mprop: "count",
          pt: "Person",
          pvs: {},
          title: "Population",
        },
      },
    });

  when(axios.get)
    .calledWith("/api/stats/stats-var-property?dcid=NotInTheTree")
    .mockResolvedValue({
      data: {
        NotInTheTree: {
          md: "",
          mprop: "count",
          pt: "Person",
          pvs: {},
          title: "",
        },
      },
    });

  // get place stats vars, geoId/05
  when(axios.post)
    .calledWith("/api/place/stat-vars/union", {
      dcids: ["geoId/05"],
      statVars: ["Median_Age_Person"],
    })
    .mockResolvedValue({
      data: ["Median_Age_Person"],
    });

  // get place stats vars, geoId/05
  when(axios.post)
    .calledWith("/api/place/stat-vars/union", {
      dcids: ["geoId/05"],
      statVars: ["Median_Age_Person", "Count_Person"],
    })
    .mockResolvedValue({
      data: ["Count_Person", "Median_Age_Person"],
    });

  // get place stats vars, geoId/05
  when(axios.post)
    .calledWith("/api/place/stat-vars/union", {
      dcids: ["geoId/05"],
      statVars: ["NotInTheTree"],
    })
    .mockResolvedValue({
      data: ["NotInTheTree"],
    });

  // get place stats vars, geoId/05
  when(axios.post)
    .calledWith("/api/place/stat-vars/union", {
      dcids: ["geoId/05"],
      statVars: ["Count_Person"],
    })
    .mockResolvedValue({
      data: ["Count_Person"],
    });

  // get place names, geoId/05
  when(axios.post)
    .calledWith("/api/place/name", {
      dcids: ["geoId/05"],
    })
    .mockResolvedValue({ data: { "geoId/05": "Place" } });

  // get data, geoId/05,Count_Person
  when(axios.get)
    .calledWith("/api/observations/series", {
      params: {
        variables: ["Count_Person"],
        entities: ["geoId/05"],
      },
      paramsSerializer: stringifyFn,
    })
    .mockResolvedValue({
      data: {
        data: {
          Count_Person: {
            "geoId/05": {
              series: [
                {
                  date: "2011",
                  value: 2690743,
                },
                {
                  date: "2012",
                  value: 2952164,
                },
              ],
              facet: "facet1",
            },
          },
          facets: {
            facet1: {
              importName: "CensusPEP",
              provenanceUrl:
                "https://www.census.gov/programs-surveys/popest.html",
              measurementMethod: "CensusPEPSurvey",
            },
          },
        },
      },
    });

  // get stats all data, geoId/05,Median_Age_Person
  when(axios.get)
    .calledWith("/api/observations/series/all", {
      params: {
        entities: ["geoId/05"],
        variables: ["Median_Age_Person"],
      },
      paramsSerializer: stringifyFn,
    })
    .mockResolvedValue({
      data: {
        data: {
          Median_Age_Person: {
            "geoId/05": [
              {
                series: [
                  {
                    date: "2011",
                    value: 37.3,
                  },
                  {
                    date: "2012",
                    value: 37.4,
                  },
                  {
                    date: "2013",
                    value: 37.5,
                  },
                  {
                    date: "2014",
                    value: 37.6,
                  },
                  {
                    date: "2015",
                    value: 37.7,
                  },
                  {
                    date: "2016",
                    value: 37.7,
                  },
                  {
                    date: "2017",
                    value: 37.9,
                  },
                  {
                    date: "2018",
                    value: 37.9,
                  },
                  {
                    date: "2019",
                    value: 38.1,
                  },
                ],
                facet: "facet1",
              },
            ],
          },
        },
        facets: {
          facet1: {
            importName: "CensusACS5YearSurvey",
            provenanceUrl: "https://www.census.gov/",
            measurementMethod: "CensusACS5yrSurvey",
            unit: "Year",
          },
        },
      },
    });

  // get stats all data, geoId/05,Count_Person
  when(axios.get)
    .calledWith("/api/observations/series/all", {
      params: {
        entities: ["geoId/05"],
        variables: ["Count_Person"],
      },
      paramsSerializer: stringifyFn,
    })
    .mockResolvedValue({
      data: {
        data: {
          Count_Person: {
            "geoId/05": [
              {
                series: [
                  {
                    date: "2001",
                    value: 2690743,
                  },
                  {
                    date: "2012",
                    value: 2952164,
                  },
                ],
                facet: "facet1",
              },
            ],
          },
        },
        facets: {
          facet1: {
            importName: "CensusPEP",
            provenanceUrl:
              "https://www.census.gov/programs-surveys/popest.html",
            measurementMethod: "CensusPEPSurvey",
          },
        },
      },
    });

  when(axios.get)
    .calledWith("/api/observations/series/all", {
      params: {
        entities: ["geoId/05"],
        variables: ["NotInTheTree"],
      },
      paramsSerializer: stringifyFn,
    })
    .mockResolvedValue({
      data: {
        data: {
          NotInTheTree: {
            "geoId/05": [
              {
                series: [
                  {
                    date: "2001",
                    value: 2690743,
                  },
                  {
                    date: "2012",
                    value: 2952164,
                  },
                ],
                facet: "facet1",
              },
            ],
          },
        },
        facets: {
          facet1: {
            importName: "CensusPEP",
            provenanceUrl:
              "https://www.census.gov/programs-surveys/popest.html",
            measurementMethod: "CensusPEPSurvey",
          },
        },
      },
    });

  when(axios.post)
    .calledWith("/api/place/displayname", { dcids: ["geoId/05"] })
    .mockResolvedValue({
      data: {
        "geoId/05": "Arkansas",
      },
    });
  when(axios.post)
    .calledWith("/api/variable-group/info", {
      dcid: "dc/g/Root",
      entities: [],
      numEntitiesExistence: undefined,
    })
    .mockResolvedValue({
      data: {
        childStatVarGroups: [
          {
            displayName: "Demographics",
            id: "dc/g/Demographics",
            specializedEntity: "Demographics",
            descendentStatVarCount: 100,
          },
          {
            displayName: "Economics",
            id: "dc/g/Economics",
            specializedEntity: "Economics",
            descendentStatVarCount: 100,
          },
        ],
      },
    });
  when(axios.post)
    .calledWith("/api/variable-group/info", {
      dcid: "dc/g/Demographics",
      entities: ["geoId/05"],
      numEntitiesExistence: undefined,
    })
    .mockResolvedValue({
      data: {
        childStatVarGroups: [
          {
            displayName: "Person By Age",
            id: "dc/g/Person_Age",
            specializedEntity: "Age",
            descendentStatVarCount: 5,
          },
          {
            displayName: "Person By ArmedForcesStatus",
            id: "dc/g/Person_ArmedForcesStatus",
            specializedEntity: "ArmedForcesStatus",
            descendentStatVarCount: 5,
          },
        ],
        childStatVars: [
          {
            displayName: "Count Of Person",
            id: "Count_Person",
            searchName: "Count Of Person",
            hasData: true,
          },
          {
            displayName: "Median age of person",
            id: "Median_Age_Person",
            searchName: "Median age of person",
            hasData: true,
          },
        ],
      },
    });
  when(axios.post)
    .calledWith("/api/variable-group/info", {
      dcid: "dc/g/Root",
      entities: ["geoId/05"],
      numEntitiesExistence: undefined,
    })
    .mockResolvedValue({
      data: {
        childStatVarGroups: [
          {
            displayName: "Demographics",
            id: "dc/g/Demographics",
            specializedEntity: "Demographics",
            descendentStatVarCount: 100,
          },
          {
            displayName: "Economics",
            id: "dc/g/Economics",
            specializedEntity: "Economics",
            descendentStatVarCount: 100,
          },
        ],
      },
    });

  when(axios.get)
    .calledWith("/api/variable/path?dcid=Count_Person")
    .mockResolvedValue({
      data: ["Count_Person", "dc/g/Demographics"],
    });

  when(axios.get)
    .calledWith("/api/variable/path?dcid=Median_Age_Person")
    .mockResolvedValue({
      data: ["Median_Age_Person", "dc/g/Demographics"],
    });

  when(axios.get)
    .calledWith("/api/variable/path?dcid=NotInTheTree")
    .mockResolvedValue({
      data: ["NotInTheTree"],
    });

  when(axios.get)
    .calledWith("/api/variable/info", {
      params: {
        dcids: ["Count_Person", "Median_Age_Person"],
      },
      paramsSerializer: stringifyFn,
    })
    .mockResolvedValue({
      data: {
        Count_Person: {
          placeTypeSummary: {
            type1: {
              numPlaces: 0,
              topPlaces: [],
            },
          },
        },
        Median_Age_Person: {
          placeTypeSummary: {
            type1: {
              numPlaces: 0,
              topPlaces: [],
            },
          },
        },
      },
    });
}

// Mock drawGroupLineChart() as getComputedTextLength can has issue with jest
// and jsdom.
export function drawGroupLineChart_mock(): void {
  (drawGroupLineChart as jest.Mock).mockImplementation(
    (selector: string | HTMLDivElement) => {
      let container: d3.Selection<any, any, any, any>;
      if (typeof selector === "string") {
        container = d3.select("#" + selector);
      } else if (selector instanceof HTMLDivElement) {
        container = d3.select(selector);
      } else {
        return;
      }
      container.selectAll("svg").remove();

      const svg = container
        .append("svg")
        .attr("width", 500)
        .attr("height", 500);
      svg.append("text").text("svg text");
    }
  );
}
