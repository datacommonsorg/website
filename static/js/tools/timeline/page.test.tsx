/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

jest.mock("axios");
jest.mock("../../chart/draw");

import { waitFor } from "@testing-library/react";
import Enzyme, { mount, ReactWrapper } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import pretty from "pretty";
import React from "react";

import { axios_mock, drawGroupLineChart_mock } from "./mock_functions";
import { Page } from "./page";

Enzyme.configure({ adapter: new Adapter() });

const globalAny: any = global;

beforeEach(() => {
  // Mock the info config object that is used for the landing page.
  window.infoConfig = [];
});

async function waitForComponentUpdates(wrapper: ReactWrapper) {
  // Wait for state updates
  await waitFor(() => {
    expect(wrapper.text()).toContain("");
  });
  // Wait for stat var info and place info fetching
  await new Promise(process.nextTick);
  wrapper.update();
  await new Promise(process.nextTick);
  wrapper.update();
}

test("Single place and single stat var", async () => {
  globalAny.window = Object.create(window);
  document.body.innerHTML =
    '<button id="download-link"></button><a id="bulk-download-link"></a>';
  // Set url hash
  Object.defineProperty(window, "location", {
    writable: true,
    value: {
      hash: "#place=geoId/05&statsVar=Median_Age_Person",
    },
  });
  // Mock drawGroupLineChart() as getComputedTextLength can has issue with jest
  // and jsdom.
  drawGroupLineChart_mock();
  // Mock all the async axios call
  axios_mock();
  // Do the actual render!
  const wrapper = mount(<Page />);
  await waitForComponentUpdates(wrapper);
  // Check that one chart is on the page and it matches snapshot
  expect(
    pretty(wrapper.find("#chart-region").getDOMNode().innerHTML)
  ).toMatchSnapshot();
  // Open demographics node in stat var widget
  wrapper
    .find("#hierarchy-section .Collapsible__trigger")
    .at(0)
    .simulate("click");
  // Wait for the stat var info fetch
  await new Promise(process.nextTick);
  wrapper.update();
  // Add one statVar by clicking the checkbox
  wrapper
    .find("#hierarchy-section input")
    .at(0)
    .simulate("change", { target: { checked: true } });
  // Check that url hash is updated
  window.location.hash = "#" + window.location.hash;
  expect(window.location.hash).toEqual(
    "#place=geoId%2F05&statsVar=Median_Age_Person__Count_Person"
  );
  // Hack to trigger hashchange event to fire
  window.dispatchEvent(
    new HashChangeEvent("hashchange", {
      newURL: "#place=geoId%2F05&statsVar=Median_Age_Person__Count_Person",
      oldURL: "#place=geoId/05&statsVar=Median_Age_Person",
    })
  );
  await waitForComponentUpdates(wrapper);
  // Check that there are 2 charts on the page and they match the snapshot
  expect(
    pretty(wrapper.find("#chart-region").getDOMNode().innerHTML)
  ).toMatchSnapshot();
  // Delete one statVar
  wrapper
    .find("#hierarchy-section input")
    .at(1)
    .simulate("change", { target: { checked: false } });
  // Check that the url hash is updated
  window.location.hash = "#" + window.location.hash;
  expect(window.location.hash).toEqual(
    "#place=geoId%2F05&statsVar=Count_Person"
  );
  // Hack to trigger hashchange event to fire
  window.dispatchEvent(
    new HashChangeEvent("hashchange", {
      newURL: "#place=geoId%2F05&statsVar=Count_Person",
      oldURL: "#place=geoId%2F05&statsVar=Median_Age_Person__Count_Person",
    })
  );
  await waitForComponentUpdates(wrapper);
  // Check that there is one chart on the page and the content matches the snapshot
  expect(
    pretty(wrapper.find("#chart-region").getDOMNode().innerHTML)
  ).toMatchSnapshot();
  // Check that the stat var widget matches the snapshot
  expect(
    pretty(wrapper.find("#hierarchy-section").getDOMNode().innerHTML)
  ).toMatchSnapshot();
});

test("statVar not in PV-tree", async () => {
  Object.defineProperty(window, "location", {
    value: {
      hash: "#&place=geoId/05&statsVar=NotInTheTree",
    },
  });
  // Mock drawGroupLineChart() as getComputedTextLength can has issue with jest
  // and jsdom.
  drawGroupLineChart_mock();
  // mock all the async axios call
  axios_mock();
  // Do the actual render!
  const wrapper = mount(<Page />);
  await waitForComponentUpdates(wrapper);
  // Check that one chart is on the page and it matches snapshot
  expect(
    pretty(wrapper.find("#chart-region").getDOMNode().innerHTML)
  ).toMatchSnapshot();
  // Check that the stat var widget matches the snapshot
  expect(
    pretty(wrapper.find("#hierarchy-section").getDOMNode().innerHTML)
  ).toMatchSnapshot();
});

test("chart options", async () => {
  globalAny.window = Object.create(window);
  Object.defineProperty(window, "location", {
    writable: true,
    value: {
      hash: "#&place=geoId/05&statsVar=Count_Person",
    },
  });

  // Mock drawGroupLineChart() as getComputedTextLength can has issue with jest
  // and jsdom.
  drawGroupLineChart_mock();
  // mock all the async axios call
  axios_mock();

  // Do the actual render!
  const wrapper = mount(<Page />);
  await waitForComponentUpdates(wrapper);
  // Set per capita to true
  wrapper.find("#count-ratio").at(0).simulate("change");
  await waitForComponentUpdates(wrapper);
  // Check that url hash is updated
  window.location.hash = "#" + window.location.hash;
  expect(window.location.hash).toBe(
    "#place=geoId%2F05&statsVar=Count_Person&chart=%7B%22count%22%3A%7B%22pc%22%3Atrue%7D%7D"
  );
  // Hack to trigger hashchange event to fire
  window.dispatchEvent(
    new HashChangeEvent("hashchange", {
      newURL:
        "#place=geoId%2F05&statsVar=Count_Person&chart=%7B%22count%22%3A%7B%22pc%22%3Atrue%7D%7D",
      oldURL: "#&place=geoId/05&statsVar=Count_Person",
    })
  );
  await waitForComponentUpdates(wrapper);
  expect(
    pretty(wrapper.find("#chart-region").getDOMNode().innerHTML)
  ).toMatchSnapshot();
  // Open the Demographics node in the stat var widget
  wrapper
    .find("#hierarchy-section .Collapsible__trigger")
    .at(0)
    .simulate("click");
  // Wait for the stat var info fetch from the stat var widget
  await new Promise(process.nextTick);
  wrapper.update();
  // Remove the stat var
  wrapper
    .find("#hierarchy-section input[checked=true]")
    .at(0)
    .simulate("change", { target: { checked: false } });
  // Check that the url hash is updated
  window.location.hash = "#" + window.location.hash;
  expect(window.location.hash).toBe(
    "#place=geoId%2F05&statsVar=&chart=%7B%22count%22%3A%7B%22pc%22%3Atrue%7D%7D"
  );
});
