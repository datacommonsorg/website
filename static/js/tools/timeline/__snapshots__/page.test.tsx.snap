// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single place and single stat var 1`] = `
"<div class=\\"chart-container\\">
  <div class=\\"card\\">
    <div class=\\"statVarChipRegion\\">
      <div class=\\"chip\\"><span class=\\"chip-text chip-clickable-text\\">Age</span><button class=\\"chip-action\\"><i class=\\"material-icons\\">cancel</i></button></div>
    </div>
    <div class=\\"chart-svg\\"><svg width=\\"500\\" height=\\"500\\"><text>svg text</text></svg></div>
  </div>
  <div class=\\"chart-footer-container no-bottom-border\\">
    <div class=\\"chart-footer-metadata-section\\">
      <div class=\\"chart-footer-metadata\\"><span>Data from <span><a href=\\"https://www.census.gov/\\">census.gov</a></span></span></div>
      <div class=\\"chart-footer-metadata\\"><span>Measurement method: CensusACS5yrSurvey</span></div>
    </div>
    <div class=\\"chart-footer-options-button\\"><span>Chart Options</span><i class=\\"material-icons\\">expand_less</i></div>
  </div>
  <div class=\\"chart-footer-options-section\\"><span class=\\"chart-option\\"><div class=\\"form-check\\"><label class=\\"form-check-label\\"><input id=\\"age-ratio\\" type=\\"checkbox\\" class=\\"form-check-input\\">Per Capita</label></div></span><span class=\\"chart-option\\"><div class=\\"form-check\\"><label class=\\"form-check-label\\"><input id=\\"delta-cb-age\\" type=\\"checkbox\\" class=\\"is-delta-input form-check-input\\">Delta</label></div></span><button type=\\"button\\" class=\\"source-selector-open-modal-button btn btn-light btn-sm\\">Edit Source</button></div>
  <div class=\\"feedback-link\\"><a href=\\"/feedback\\">Feedback</a></div>
</div>"
`;

exports[`Single place and single stat var 2`] = `
"<div class=\\"chart-container\\">
  <div class=\\"card\\">
    <div class=\\"statVarChipRegion\\">
      <div class=\\"chip\\"><span class=\\"chip-text chip-clickable-text\\">Population</span><button class=\\"chip-action\\"><i class=\\"material-icons\\">cancel</i></button></div>
    </div>
    <div class=\\"chart-svg\\"><svg width=\\"500\\" height=\\"500\\"><text>svg text</text></svg></div>
  </div>
  <div class=\\"chart-footer-container no-bottom-border\\">
    <div class=\\"chart-footer-metadata-section\\">
      <div class=\\"chart-footer-metadata\\"><span>Data from <span><a href=\\"https://www.census.gov/programs-surveys/popest.html\\">census.gov</a></span></span></div>
      <div class=\\"chart-footer-metadata\\"><span>Measurement method: CensusPEPSurvey</span></div>
    </div>
    <div class=\\"chart-footer-options-button\\"><span>Chart Options</span><i class=\\"material-icons\\">expand_less</i></div>
  </div>
  <div class=\\"chart-footer-options-section\\"><span class=\\"chart-option\\"><div class=\\"form-check\\"><label class=\\"form-check-label\\"><input id=\\"count-ratio\\" type=\\"checkbox\\" class=\\"form-check-input\\">Per Capita</label></div></span><span class=\\"chart-option\\"><div class=\\"form-check\\"><label class=\\"form-check-label\\"><input id=\\"delta-cb-count\\" type=\\"checkbox\\" class=\\"is-delta-input form-check-input\\">Delta</label></div></span><button type=\\"button\\" class=\\"source-selector-open-modal-button btn btn-light btn-sm\\">Edit Source</button></div>
  <div class=\\"feedback-link\\"><a href=\\"/feedback\\">Feedback</a></div>
</div>
<div class=\\"chart-container\\">
  <div class=\\"card\\">
    <div class=\\"statVarChipRegion\\">
      <div class=\\"chip\\"><span class=\\"chip-text chip-clickable-text\\">Age</span><button class=\\"chip-action\\"><i class=\\"material-icons\\">cancel</i></button></div>
    </div>
    <div class=\\"chart-svg\\"><svg width=\\"500\\" height=\\"500\\"><text>svg text</text></svg></div>
  </div>
  <div class=\\"chart-footer-container no-bottom-border\\">
    <div class=\\"chart-footer-metadata-section\\">
      <div class=\\"chart-footer-metadata\\"><span>Data from <span><a href=\\"https://www.census.gov/\\">census.gov</a></span></span></div>
      <div class=\\"chart-footer-metadata\\"><span>Measurement method: CensusACS5yrSurvey</span></div>
    </div>
    <div class=\\"chart-footer-options-button\\"><span>Chart Options</span><i class=\\"material-icons\\">expand_less</i></div>
  </div>
  <div class=\\"chart-footer-options-section\\"><span class=\\"chart-option\\"><div class=\\"form-check\\"><label class=\\"form-check-label\\"><input id=\\"age-ratio\\" type=\\"checkbox\\" class=\\"form-check-input\\">Per Capita</label></div></span><span class=\\"chart-option\\"><div class=\\"form-check\\"><label class=\\"form-check-label\\"><input id=\\"delta-cb-age\\" type=\\"checkbox\\" class=\\"is-delta-input form-check-input\\">Delta</label></div></span><button type=\\"button\\" class=\\"source-selector-open-modal-button btn btn-light btn-sm\\">Edit Source</button></div>
  <div class=\\"feedback-link\\"><a href=\\"/feedback\\">Feedback</a></div>
</div>"
`;

exports[`Single place and single stat var 3`] = `
"<div class=\\"chart-container\\">
  <div class=\\"card\\">
    <div class=\\"statVarChipRegion\\">
      <div class=\\"chip\\"><span class=\\"chip-text chip-clickable-text\\">Population</span><button class=\\"chip-action\\"><i class=\\"material-icons\\">cancel</i></button></div>
    </div>
    <div class=\\"chart-svg\\"><svg width=\\"500\\" height=\\"500\\"><text>svg text</text></svg></div>
  </div>
  <div class=\\"chart-footer-container no-bottom-border\\">
    <div class=\\"chart-footer-metadata-section\\">
      <div class=\\"chart-footer-metadata\\"><span>Data from <span><a href=\\"https://www.census.gov/programs-surveys/popest.html\\">census.gov</a></span></span></div>
      <div class=\\"chart-footer-metadata\\"><span>Measurement method: CensusPEPSurvey</span></div>
    </div>
    <div class=\\"chart-footer-options-button\\"><span>Chart Options</span><i class=\\"material-icons\\">expand_less</i></div>
  </div>
  <div class=\\"chart-footer-options-section\\"><span class=\\"chart-option\\"><div class=\\"form-check\\"><label class=\\"form-check-label\\"><input id=\\"count-ratio\\" type=\\"checkbox\\" class=\\"form-check-input\\">Per Capita</label></div></span><span class=\\"chart-option\\"><div class=\\"form-check\\"><label class=\\"form-check-label\\"><input id=\\"delta-cb-count\\" type=\\"checkbox\\" class=\\"is-delta-input form-check-input\\">Delta</label></div></span><button type=\\"button\\" class=\\"source-selector-open-modal-button btn btn-light btn-sm\\">Edit Source</button></div>
  <div class=\\"feedback-link\\"><a href=\\"/feedback\\">Feedback</a></div>
</div>"
`;

exports[`Single place and single stat var 4`] = `
"<div class=\\"Collapsible\\"><span class=\\"Collapsible__trigger is-open\\"><div class=\\"node-title\\"><i class=\\"material-icons\\">remove</i><span class=\\"title selected-node-title\\">Demographics<span class=\\"sv-count\\">(100)</span></span></div></span>
<div class=\\"Collapsible__contentOuter\\" style=\\"height: 0px; transition: height 200ms linear; overflow: hidden;\\">
  <div class=\\"Collapsible__contentInner\\">
    <div class=\\"svg-node-child\\">
      <div>
        <form class=\\"node-title\\"><input id=\\"Count_Persondc/g/Demographics-Count_Person\\" name=\\"stat-var-hierarchy\\" type=\\"checkbox\\"><label class=\\"selected-node-title\\" for=\\"Count_Persondc/g/Demographics-Count_Person\\">Count Of Person</label></form>
      </div>
      <div>
        <form class=\\"node-title\\"><input id=\\"Median_Age_Persondc/g/Demographics-Median_Age_Person\\" name=\\"stat-var-hierarchy\\" type=\\"checkbox\\"><label class=\\"\\" for=\\"Median_Age_Persondc/g/Demographics-Median_Age_Person\\">Median age of person</label></form>
      </div>
    </div>
    <div class=\\"svg-node-child\\">
      <div>
        <div class=\\"Collapsible\\"><span class=\\"Collapsible__trigger is-closed\\"><div class=\\"node-title\\"><i class=\\"material-icons\\">add</i><span class=\\"title\\">Age<span class=\\"sv-count\\">(5)</span></span></div></span>
        <div class=\\"Collapsible__contentOuter\\" style=\\"height: 0px; transition: height 200ms linear; overflow: hidden;\\">
          <div class=\\"Collapsible__contentInner\\"></div>
        </div>
      </div>
    </div>
    <div>
      <div class=\\"Collapsible\\"><span class=\\"Collapsible__trigger is-closed\\"><div class=\\"node-title\\"><i class=\\"material-icons\\">add</i><span class=\\"title\\">ArmedForcesStatus<span class=\\"sv-count\\">(5)</span></span></div></span>
      <div class=\\"Collapsible__contentOuter\\" style=\\"height: 0px; transition: height 200ms linear; overflow: hidden;\\">
        <div class=\\"Collapsible__contentInner\\"></div>
      </div>
    </div>
  </div>
</div>
</div>
</div>
</div>
<div class=\\"Collapsible\\"><span class=\\"Collapsible__trigger is-closed\\"><div class=\\"node-title\\"><i class=\\"material-icons\\">add</i><span class=\\"title\\">Economics<span class=\\"sv-count\\">(100)</span></span></div></span>
<div class=\\"Collapsible__contentOuter\\" style=\\"height: 0px; transition: height 200ms linear; overflow: hidden;\\">
  <div class=\\"Collapsible__contentInner\\"></div>
</div>
</div>"
`;

exports[`chart options 1`] = `
"<div class=\\"chart-container\\">
  <div class=\\"card\\">
    <div class=\\"statVarChipRegion\\">
      <div class=\\"chip\\"><span class=\\"chip-text chip-clickable-text\\">Population</span><button class=\\"chip-action\\"><i class=\\"material-icons\\">cancel</i></button></div>
    </div>
    <div class=\\"chart-svg\\"><svg width=\\"500\\" height=\\"500\\"><text>svg text</text></svg></div>
  </div>
  <div class=\\"chart-footer-container no-bottom-border\\">
    <div class=\\"chart-footer-metadata-section\\">
      <div class=\\"chart-footer-metadata\\"><span>Data from <span><a href=\\"https://www.census.gov/programs-surveys/popest.html\\">census.gov</a></span></span></div>
      <div class=\\"chart-footer-metadata\\"><span>Measurement method: CensusPEPSurvey</span></div>
    </div>
    <div class=\\"chart-footer-options-button\\"><span>Chart Options</span><i class=\\"material-icons\\">expand_less</i></div>
  </div>
  <div class=\\"chart-footer-options-section\\"><span class=\\"chart-option\\"><div class=\\"form-check\\"><label class=\\"form-check-label\\"><input id=\\"count-ratio\\" type=\\"checkbox\\" class=\\"form-check-input\\">Per Capita</label></div></span><span class=\\"chart-option\\"><div class=\\"form-check\\"><label class=\\"form-check-label\\"><input id=\\"delta-cb-count\\" type=\\"checkbox\\" class=\\"is-delta-input form-check-input\\">Delta</label></div></span><button type=\\"button\\" class=\\"source-selector-open-modal-button btn btn-light btn-sm\\">Edit Source</button></div>
  <div class=\\"feedback-link\\"><a href=\\"/feedback\\">Feedback</a></div>
</div>"
`;

exports[`statVar not in PV-tree 1`] = `
"<div class=\\"chart-container\\">
  <div class=\\"card\\">
    <div class=\\"statVarChipRegion\\">
      <div class=\\"chip\\"><span class=\\"chip-text chip-clickable-text\\"></span><button class=\\"chip-action\\"><i class=\\"material-icons\\">cancel</i></button></div>
    </div>
    <div class=\\"chart-svg\\"><svg width=\\"500\\" height=\\"500\\"><text>svg text</text></svg></div>
  </div>
  <div class=\\"chart-footer-container no-bottom-border\\">
    <div class=\\"chart-footer-metadata-section\\">
      <div class=\\"chart-footer-metadata\\"><span>Data from <span><a href=\\"https://www.census.gov/programs-surveys/popest.html\\">census.gov</a></span></span></div>
      <div class=\\"chart-footer-metadata\\"><span>Measurement method: CensusPEPSurvey</span></div>
    </div>
    <div class=\\"chart-footer-options-button\\"><span>Chart Options</span><i class=\\"material-icons\\">expand_less</i></div>
  </div>
  <div class=\\"chart-footer-options-section\\"><span class=\\"chart-option\\"><div class=\\"form-check\\"><label class=\\"form-check-label\\"><input id=\\"count-ratio\\" type=\\"checkbox\\" class=\\"form-check-input\\">Per Capita</label></div></span><span class=\\"chart-option\\"><div class=\\"form-check\\"><label class=\\"form-check-label\\"><input id=\\"delta-cb-count\\" type=\\"checkbox\\" class=\\"is-delta-input form-check-input\\">Delta</label></div></span><button type=\\"button\\" class=\\"source-selector-open-modal-button btn btn-light btn-sm\\">Edit Source</button></div>
  <div class=\\"feedback-link\\"><a href=\\"/feedback\\">Feedback</a></div>
</div>"
`;

exports[`statVar not in PV-tree 2`] = `
"<div class=\\"Collapsible\\"><span class=\\"Collapsible__trigger is-closed\\"><div class=\\"node-title\\"><i class=\\"material-icons\\">add</i><span class=\\"title\\">Demographics<span class=\\"sv-count\\">(100)</span></span></div></span>
<div class=\\"Collapsible__contentOuter\\" style=\\"height: 0px; transition: height 200ms linear; overflow: hidden;\\">
  <div class=\\"Collapsible__contentInner\\"></div>
</div>
</div>
<div class=\\"Collapsible\\"><span class=\\"Collapsible__trigger is-closed\\"><div class=\\"node-title\\"><i class=\\"material-icons\\">add</i><span class=\\"title\\">Economics<span class=\\"sv-count\\">(100)</span></span></div></span>
<div class=\\"Collapsible__contentOuter\\" style=\\"height: 0px; transition: height 200ms linear; overflow: hidden;\\">
  <div class=\\"Collapsible__contentInner\\"></div>
</div>
</div>"
`;
